### PR TITLE
Log local position setpoint reference for fixedwings when running NPFG

### PR DESCRIPTION
**Pull request by Jaeyoung-Lim**
## Describe problem solved by this pull request
For fixedwing position control, waypoint coordinates projected on to the local frame are logged as local position setpoints. While this makes sense for waypoints, NPFG is capable of tracking paths, and logging what position reference the vehicle is trying to track would be much more informative. For example, when the vehicle is flying on a loiter path, we want to know how close the vehicle is to the closest point on the circle, rather than the center of the loiter circle.

## Describe your solution
This commit adds a state in the NPFG library which keeps the position setpoint reference so that it can be published as part of the `vehicle_local_position_setpoint` message.

When passing arbitrary references with a path in offboard mode, comparing the position reference and the current position provides better insights into how well the vehicle is tracking the path. 

## Test data / coverage
Tested in SITL Gazebo
```
make px4_sitl gazebo_plane
```

**Before PR**
- Flight log: [log](https://review.px4.io/plot_app?log=1f580ba6-8e1a-4c79-886a-76215c045744)
bokeh_plot (6)
bokeh_plot (7)

**After PR**
- Flight log: [log](https://review.px4.io/plot_app?log=c1b856a9-3240-432e-bac8-0570388a6c95)
- Part of the log shows a continuous position references from a loiter waypoint
bokeh_plot (4)
bokeh_plot (5)

## Additional context
Add any other related context or media.

### CODE REVIEW

File: src/lib/npfg/npfg.hpp [hunk 1]
```diff
@@ -422,6 +424,7 @@ class NPFG
 	matrix::Vector2f unit_path_tangent_{matrix::Vector2f{1.0f, 0.0f}}; // unit path tangent vector
 	float signed_track_error_{0.0f}; // signed track error [m]
 	matrix::Vector2f bearing_vec_{matrix::Vector2f{1.0f, 0.0f}}; // bearing unit vector
+	matrix::Vector2f position_setpoint_{matrix::Vector2f{NAN, NAN}};
```
Comment (tstastny) [inline src/lib/npfg/npfg.hpp:427]: I would either rename this the "closest point", or at least put that the "position setpoint" corresponds to the closest point from the vehicle on the path. Just to make a clear difference between "position setpoints" getting put in as waypoints.. and what the guidance law is actually projecting/tracking on the path.

```suggestion
	matrix::Vector2f closest_point_on_path_{matrix::Vector2f{NAN, NAN}}; // instantaneous position setpoint [m]
```

**Review by tstastny — CHANGES_REQUESTED** [review]
Nice idea to log this. A few corrections / comments. Also - @sfuhrer maybe you could comment on if publishing the local position setpoint topic doesnt interfere with e.g. multicopter control when the aircraft is a VTOL and running both controllers?

@Jaeyoung-Lim can you put a plot of the setpoint working for waypoint following as well after making these changes? -- just so we can see that it does what we think it does. There would be a jump most likely when switching from flying directly to waypoint to following line.

**Comment by Jaeyoung-Lim** [comment]
@tstastny Thanks for the review and corrections!

Here is the log with a normal waypoint navigation: https://review.px4.io/plot_app?log=76b7eeaa-db7f-4813-ae06-d6cf32cad181
bokeh_plot (9)
bokeh_plot (8)

**Review by tstastny — APPROVED** [review]

**Comment by sfuhrer** [comment]
>Also - @sfuhrer maybe you could comment on if publishing the local position setpoint topic doesnt interfere with e.g. multicopter control when the aircraft is a VTOL and running both controllers?

I was also published before, so I don't see an issue.

**Pull request merged**